In [3]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline

In [4]:
dataset = pd.read_csv(filepath_or_buffer='Data/TRN.csv', sep='\t', index_col = 0)

<h1>Basic Informations</h1>

In [21]:
dataset.head()

UF_1  UF_2  UF_3  UF_4  UF_5  UF_6  UF_7     IDADE  SEXO_1  \
INDEX                                                               
0         1     1     1     0     0     0     0  0.135098       1   
1         1     0     1     0     0     1     0  0.273504       1   
2         1     0     1     0     0     1     0  0.281910       0   
3         1     1     1     0     0     0     0  0.225741       0   
4         1     1     0     0     0     1     0  0.480403       0   

       NIVEL_RELACIONAMENTO_CREDITO01     ...       CEP4_6  CEP4_7  CEP4_8  \
INDEX                                     ...                                
0                            0.222222     ...            0       0       0   
1                            0.111111     ...            0       0       1   
2                            1.000000     ...            0       1       1   
3                            0.111111     ...            0       1       1   
4                            0.111111     ...            1       1       1   

       CEP4_9  CEP4_10  CEP4_11  CEP4_12  CEP4_13  CEP4_14  IND_BOM_1_1  
INDEX                                                                    
0           1        1        0        1        1        1            0  
1           0        1        1        0        0        0            1  
2           0        0        0        0        1        0            1  
3           0        1        1        0        1        0            1  
4           1        0        0        1        0        1            1  

[5 rows x 244 columns]

In [23]:
dataset.head().to_csv('dataHead.csv')

In [6]:
dataset.shape

(389196, 245)

In [7]:
dataset.describe()

UF_1           UF_2           UF_3           UF_4  \
count  389196.000000  389196.000000  389196.000000  389196.000000   
mean        0.889274       0.691952       0.476552       0.296195   
std         0.313793       0.461687       0.499451       0.456579   
min         0.000000       0.000000       0.000000       0.000000   
25%         1.000000       0.000000       0.000000       0.000000   
50%         1.000000       1.000000       0.000000       0.000000   
75%         1.000000       1.000000       1.000000       1.000000   
max         1.000000       1.000000       1.000000       1.000000   

                UF_5           UF_6           UF_7         IDADE  \
count  389196.000000  389196.000000  389196.000000  3.891960e+05   
mean        0.241179       0.218011       0.186836  4.552049e-01   
std         0.427799       0.412895       0.389781  2.537459e-01   
min         0.000000       0.000000       0.000000  5.506237e-16   
25%         0.000000       0.000000       0.000000  2.507866e-01   
50%         0.000000       0.000000       0.000000  4.375241e-01   
75%         0.000000       0.000000       0.000000  6.578835e-01   
max         1.000000       1.000000       1.000000  1.000000e+00   

              SEXO_1  NIVEL_RELACIONAMENTO_CREDITO01      ...        \
count  389196.000000                   389196.000000      ...         
mean        0.521514                        0.134915      ...         
std         0.499538                        0.133136      ...         
min         0.000000                        0.000000      ...         
25%         0.000000                        0.111111      ...         
50%         1.000000                        0.111111      ...         
75%         1.000000                        0.111111      ...         
max         1.000000                        1.000000      ...         

              CEP4_7         CEP4_8         CEP4_9        CEP4_10  \
count  389196.000000  389196.000000  389196.000000  389196.000000   
mean        0.423378       0.417540       0.425708       0.459820   
std         0.494095       0.493154       0.494451       0.498384   
min         0.000000       0.000000       0.000000       0.000000   
25%         0.000000       0.000000       0.000000       0.000000   
50%         0.000000       0.000000       0.000000       0.000000   
75%         1.000000       1.000000       1.000000       1.000000   
max         1.000000       1.000000       1.000000       1.000000   

             CEP4_11        CEP4_12        CEP4_13        CEP4_14  \
count  389196.000000  389196.000000  389196.000000  389196.000000   
mean        0.440842       0.436896       0.433709       0.440339   
std         0.496489       0.496002       0.495587       0.496428   
min         0.000000       0.000000       0.000000       0.000000   
25%         0.000000       0.000000       0.000000       0.000000   
50%         0.000000       0.000000       0.000000       0.000000   
75%         1.000000       1.000000       1.000000       1.000000   
max         1.000000       1.000000       1.000000       1.000000   

         IND_BOM_1_1    IND_BOM_1_2  
count  389196.000000  389196.000000  
mean        0.655449       0.344551  
std         0.475222       0.475222  
min         0.000000       0.000000  
25%         0.000000       0.000000  
50%         1.000000       0.000000  
75%         1.000000       1.000000  
max         1.000000       1.000000  

[8 rows x 245 columns]

In [24]:
dataset.describe().to_csv('dataDescribe.csv')

<p>As we can see, we have some basic informations:</p>
<ul>
    <li>Shape: (389196, 245)</li>
    <li>Classes: IND_BOM_1_1(~66%) and IND_BOM_1_2(~34%)</li>
</ul>

<h2>Classes (IND_BOM_1_1 and IND_BOM_1_2)</h2>
<p>Here we can join both columns and define IND_BOM_1_1 as 1 and IND_BOM_1_2 as 0. For that we just need to drop the column  IND_BOM_1_1.</p>

In [8]:
dataset = dataset.drop(['IND_BOM_1_2'], axis=1)

In [ ]:
dataset.cov

In [ ]:
correlation_matrix = dataset.corr()

In [ ]:
correlation_matrix['IND_BOM_1_1'][(correlation_matrix['IND_BOM_1_1'] > 0.05) & (correlation_matrix['IND_BOM_1_1'] < 1.0)]

# Identify categorical atributes

In [19]:
dataset.dtypes.to_csv('dataTypes')